app/services/hyrax/analytics/google.rb

```diff
@@ -0,0 +1,217 @@
+# frozen_string_literal: true
+
+require 'oauth2'
+require 'signet/oauth_2/client'
+
+# OVERRIDE: Hyrax hyrax-v3.5.0 to make analytics a bit more dynamic
+# TODO(alishaevn): remove once https://github.com/samvera/hyrax/pull/6063 is available in our current hyrax version
+
+module Hyrax
+  module Analytics
+    module Google
+      extend ActiveSupport::Concern
+      # rubocop:disable Metrics/BlockLength
+      class_methods do
+        # Loads configuration options from config/analytics.yml for Hyrax apps only. You only need PRIVATE_KEY_PATH or
+        # PRIVATE_KEY_VALUE. VALUE takes precedence.
+        # Expected structure:
+        # `analytics:`
+        # `  google:`
+        # `    app_name: <%= ENV['GOOGLE_OAUTH_APP_NAME']`
+        # `    app_version: <%= ENV['GOOGLE_OAUTH_APP_VERSION']`
+        # `    privkey_path: <%= ENV['GOOGLE_OAUTH_PRIVATE_KEY_PATH']`
+        # `    privkey_value: <%= ENV['GOOGLE_OAUTH_PRIVATE_KEY_VALUE']`
+        # `    privkey_secret: <%= ENV['GOOGLE_OAUTH_PRIVATE_KEY_SECRET']`
+        # `    client_email: <%= ENV['GOOGLE_OAUTH_CLIENT_EMAIL']`
+        # @return [Config]
+        def config
+          @config ||= Config.load_from_yaml
+        end
+
+        class Config
+          def self.load_from_yaml
+            filename = Rails.root.join('config', 'analytics.yml')
+            yaml = YAML.safe_load(ERB.new(File.read(filename)).result)
+            unless yaml
+              Hyrax.logger.error("Unable to fetch any keys from #{filename}.")
+              return new({})
+            end
+            config = yaml.fetch('analytics')&.fetch('google', nil)
+            unless config
+              Deprecation.warn("Deprecated analytics configuration format found. Please update config/analytics.yml.")
+              config = yaml.fetch('analytics')
+              # this has to exist here with a placeholder so it can be set in the Hyrax initializer
+              # it is only for backward compatibility
+              config['analytics_id'] = '-'
+            end
+            new config
+          end
+
+          # OVERRIDE: Hyrax hyrax-v3.5.0 to add the KEYS variable
+          KEYS = %w[analytics_id app_name app_version privkey_path privkey_value privkey_secret client_email].freeze
+          REQUIRED_KEYS = %w[analytics_id app_name app_version privkey_secret client_email].freeze
+
+          def initialize(config)
+            @config = config
+          end
+
+          # @return [Boolean] are all the required values present?
+          def valid?
+            config_keys = @config.keys
+            # OVERRIDE: Hyrax hyrax-v3.5.0 to require either setting
+            return false unless config_keys.include?('privkey_value') || config_keys.include?('privkey_path')
+
+            REQUIRED_KEYS.all? { |required| config_keys.include?(required) && @config[required].present? }
```

Comment:
```suggestion
            REQUIRED_KEYS.all? { |required| @config[required].present? }
```

Reply:
actually I think the config_keys.include? is reduant. its either present or missing or blank and we don't care which of the later two it is.